### PR TITLE
Feature: File models

### PR DESCRIPTION
**Pull request by aaxelb**
## Ticket
https://openscience.atlassian.net/browse/EOSF-6

# Purpose
Implement file model, add file detail route and a basic file tree to the dummy app.

# Notes for Reviewers
Mirroring an oddity in the api, `node.files` is actually a list of `file-provider`s, and `fileProvider.files` contains actual `file`s. 

### Routes Added/Updated
`file/:file_id`

### CODE REVIEW

File: tests/dummy/app/templates/components/file-tree.hbs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div style="padding-left:1em">
```
Comment (samchrisinger) [inline tests/dummy/app/templates/components/file-tree.hbs]: @aaxelb please don't get carried away with this implementation. A more full-featured file tree will happen: https://openscience.atlassian.net/browse/EOSF-16

Comment (samchrisinger) [inline tests/dummy/app/templates/components/file-tree.hbs]: Also please keep css out of handlebars templates, instead use a style.css in this folder https://github.com/CenterForOpenScience/ember-osf/blob/develop/.github/CONTRIBUTING.md#styling-components

Comment (aaxelb) [inline tests/dummy/app/templates/components/file-tree.hbs]: Makes sense, thanks

File: addon/adapters/file-provider.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import ApplicationAdapter from './application';
+
+export default ApplicationAdapter.extend({
```
Comment (samchrisinger) [inline addon/adapters/file-provider.js]: Is this adapter needed? The superclass impl. should already return 'files': https://github.com/emberjs/data/blob/v2.5.2/addon/adapters/json-api.js#L110

Comment (aaxelb) [inline addon/adapters/file-provider.js]: Won't it return 'fileProviders'? The /files endpoint returns objects that should be treated as `file-provider`s, is the problem.

Comment (samchrisinger) [inline addon/adapters/file-provider.js]: Oh, good call. Thanks for the clarification.

**Pull request merged**